ERCS/erc-7585.md

```diff
@@ -0,0 +1,292 @@
+
+The following is the pseudo code for generating Mixhash:
+
+```solidity
+function generateMixHash(blockHeight,hashType,file) {
+    for chunk in file {
+      if chunk.length < 1024 {
+        chunk.pad(1024-chunk.length,0);
+      }
+      chunk_hash_array.append(getChunkHash(chunk,hashType));
+    }
+
+    merkle_tree_root = getMerkleTreeRoot(chunk_hash_array,hash_type);
+    return mix_hash(hash_type, file.length, merkle_tree_root);
+}
+```
+
+### Public Data Storage Proofs
+
+When MixHash is used to identify a piece of public data, anyone can construct a storage proof to demonstrate possession of a copy of that data. Here is a typical process for using a public data storage proof:
+
+0. Users eligible to submit storage proofs for rewards are referred to as Suppliers.
+1. A Supplier prepares a storage proof for data D (with MixHash `mix_hash_d`) based on a block at height `h`. A 256-bit `nonce` value for the proof is derived from this block (usually directly using the block's hash).
+2. To generate a correct storage proof, the Supplier needs to traverse every 1KB chunk of D to find the optimal leaf node `m`. This is done by attempting to append the nonce value to the end of each chunk to minimize the new Merkle tree root hash. After determining `m`, the path `m_path` and leaf node value `m_leaf_data` of `m` are extracted.
+3. The Supplier constructs the storage proof for data D at block time `h` using `{mix_hash_d, h, m, m_path, m_leaf_data}` and submits it to the public network.
+4. The public network can validate the correctness of `m`, `m_path`, and `m_leaf_data` based on `mix_hash_d`: verifying that `m` is indeed a chunk of D. The timeliness of the proof can be verified through `h`. After passing both correctness and timeliness checks, the public network calculates `proof_result_m` based on the nonce value and existing proof information, and saves it.
+5. The public network does not have enough information to verify the optimality of the proof, but other Suppliers with the full data set can submit a better `{mix_hash_d, h, better_m, better_m_path, better_m_leaf_data}` to challenge the published storage proof.
+6. The public network can determine the success of the challenge by comparing `proof_result_m` and `proof_result_better_m`. A successful challenge indicates the old storage proof was forged. If no one challenges the published storage proof within a certain timeframe, it can be considered correct from a game-theoretic perspective.
+7. To support healthy competition, the public network should design an appropriate economic model, rewarding users who provide correct storage proofs and penalizing those who submit false ones.
+
+With an understanding of the above process, let us describe the generation and verification of storage proofs more precisely using pseudocode.
+
+```solidity
+function generateProof(mixHash, blockHeight,file) {
```

Comment:
This function `generateProof` shall move to the "Reference Implementation" section

Reply:
This is not a complete implementation, but a pseudocode intended to describe the process more accurately. Perhaps I can write the pseudocode to be more akin to English.